third_party/move/move-core/types/src/value.rs

```diff
@@ -28,18 +28,28 @@ pub const MOVE_STRUCT_TYPE: &str = "type";
 /// In the `WithTypes` configuration, a Move struct gets serialized into a Serde struct with this as the second field
 pub const MOVE_STRUCT_FIELDS: &str = "fields";
 
+/// In the `WithVariant` configuration, a Move enum variant gets serialized into a Serde struct with this name
+pub const MOVE_VARIANT_NAME: &str = "variant";
+
+/// In the `WithVariant` configuration, a Move enum variant gets serialized into a Serde struct with this as the first field
+pub const MOVE_VARIANT_NAME_FIELD: &str = "name";
+
 #[derive(Debug, PartialEq, Eq, Clone)]
 #[cfg_attr(any(test, feature = "fuzzing"), derive(arbitrary::Arbitrary))]
 pub enum MoveStruct {
     /// The representation used by the MoveVM
     Runtime(Vec<MoveValue>),
+    /// The representation used by the MoveVM for a variant value.
+    RuntimeVariant(u16, Vec<MoveValue>),
     /// A decorated representation with human-readable field names
     WithFields(Vec<(Identifier, MoveValue)>),
     /// An even more decorated representation with both types and human-readable field names
     WithTypes {
         type_: StructTag,
         fields: Vec<(Identifier, MoveValue)>,
     },
+    /// A decorated representation of a variant, with the variant name, tag value, and field values.
+    WithVariant(Identifier, u16, Vec<(Identifier, MoveValue)>),
```

Comment:
nit: `WithVariantFields`? Otherwise a bit confusing that we have `Runtime`/`Runtime` but then `WithFields`/`WithFields`.